frontend/src/app/tag/document-tags/document-tags.component.ts

```diff
@@ -0,0 +1,44 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { FoundDocument, Tag } from '../../models';
+import { faPlus, faTimes } from '@fortawesome/free-solid-svg-icons';
+import { first, map, mergeMap } from 'rxjs/operators';
+import * as _ from 'lodash';
+
+@Component({
+  selector: 'ia-document-tags',
+  templateUrl: './document-tags.component.html',
+  styleUrls: ['./document-tags.component.scss']
+})
+export class DocumentTagsComponent implements OnInit {
+    @Input() document: FoundDocument;
+
+    faTimes = faTimes;
+    faPlus = faPlus;
+
+    showAddNew = false;
+
+    constructor() { }
+
+    ngOnInit(): void {
+    }
+
+    addTag(tagId: number) {
+        const op = (ids: number[]) => ids.concat([tagId]);
+        this.setTags(op);
+    }
+
+    removeTag(tag: Tag) {
+        const op = (ids: number[]) => ids.filter(id => id !== tag.id);
+        this.setTags(op);
+    }
+
+    private setTags(operation: (ids: number[]) => number[]) {
+        this.document.tags$.pipe(
+            first(),
```

Comment:
```suggestion
        this.document.tags$.pipe(
            takeUntil(this.destroy$),
            first(),
```
This has potential to be leaky